compiler/grainformat/comment_utils.re

```diff
@@ -0,0 +1,519 @@
+open Grain_diagnostics;
+module Doc = Res_doc;
+
+let rec get_comments_on_line = (line: int, comments) =>
+  if (List.length(comments) == 0) {
```

Comment:
It looks to me like this function is begging to use a pattern match!

Reply:
It is, and that's done

Reply:
And done throughout